### PR TITLE
InsertFinalizedDeposits: return an error

### PR DESCRIPTION
**Pull request by saolyn**
**What type of PR is this?**

Error handling

**What does this PR do? Why is it needed?**

Adds the error return type to `InsertFinalizedDeposits` and modifies it's usages to handle the error. 
We modified the code in the new deposit tree to return the errors.

**Other notes for review**

### CODE REVIEW

File: beacon-chain/blockchain/process_block_test.go [hunk 1]
```diff
@@ -981,7 +981,8 @@ func TestInsertFinalizedDeposits_MultipleFinalizedRoutines(t *testing.T) {
 		}, Proof: [][]byte{root}}, 100+i, int64(i), bytesutil.ToBytes32(root)))
 	}
 	// Insert 3 deposits before hand.
-	depositCache.InsertFinalizedDeposits(ctx, 2)
+	err = depositCache.InsertFinalizedDeposits(ctx, 2)
+	require.NoError(t, err)
```
Comment (rkapka) [inline beacon-chain/blockchain/process_block_test.go:985]: You can write this and the other ones in one line:
```
require.NoError(t, depositCache.InsertFinalizedDeposits(ctx, 2))
```

File: beacon-chain/cache/depositcache/deposits_cache.go [hunk 2]
```diff
@@ -162,12 +162,12 @@ func (dc *DepositCache) InsertFinalizedDeposits(ctx context.Context, eth1Deposit
 		}
 		depHash, err := d.Deposit.Data.HashTreeRoot()
 		if err != nil {
-			log.WithError(err).Error("Could not hash deposit data. Finalized deposit cache not updated.")
-			return
+			//log.WithError(err).Error("Could not hash deposit data. Finalized deposit cache not updated.")
```
Comment (rkapka) [inline beacon-chain/cache/depositcache/deposits_cache.go:165]: please remove comment

File: beacon-chain/cache/depositcache/deposits_cache.go [hunk 3]
```diff
@@ -162,12 +162,12 @@ func (dc *DepositCache) InsertFinalizedDeposits(ctx context.Context, eth1Deposit
...
+			return errors.Wrap(err, "Could not hash deposit data. Finalized deposit cache not updated.")
```
Comment (rkapka) [inline beacon-chain/cache/depositcache/deposits_cache.go:166]: Errors should be lowercase and without punctuation. A better message would be `could not hash deposit data`

File: beacon-chain/cache/depositcache/deposits_cache.go [hunk 4]
```diff
@@ -162,12 +162,12 @@ func (dc *DepositCache) InsertFinalizedDeposits(ctx context.Context, eth1Deposit
...
 		}
 		if err = depositTrie.Insert(depHash[:], insertIndex); err != nil {
-			log.WithError(err).Error("Could not insert deposit hash")
-			return
+			//log.WithError(err).Error("Could not insert deposit hash")
+			return errors.Wrap(err, "Could not insert deposit hash")
```
Comment (rkapka) [inline beacon-chain/cache/depositcache/deposits_cache.go:170]: errors should be lowercase and without punctuation

**Review by rkapka — CHANGES_REQUESTED** [review]

**Comment by rauljordan** [comment]
hi @rkapka this should be ready again

**Review by rkapka — APPROVED** [review]

**Pull request merged**